torch/_dynamo/variables/iter.py

```diff
@@ -97,5 +111,5 @@ def next_variables(self, tx):
             tx.replace_all(self, next_iter)
             return self.item.add_options(self), next_iter
         else:
-            raise StopIteration
+            raise InlinedUserStopIteration()
```

Comment:
I'm not auditing that all of these sites were updated. If you want me to audit it, split this refactor into a prep refactor PR.

Reply:
I've checked that all the dynamo raises of `StopIteration` have been replaced in this PR.

Reply:
To be comprehensive:

We raise `InlinedUserStopIteration` if:
1. User code raises `StopIteration`
2. List iterator escapes
3. CycleIterator escapes (cycle is trivial)

We catch `InlinedUserStopIteration` if:
1. Calling next from within a `FOR_ITER` - jump to exit
2. Calling next from within a `YIELD_FROM` - exit yield
3. CycleIterator's underlying iterator escapes - we then repeat the saved vales
4. `InliningInstructionTranslator` has `allow_stopiteration=True`
5. Uncaught when running `step`